src/SprykerEco/Zed/Inxmail/Business/Mapper/Order/AbstractOrderMapper.php

```diff
@@ -0,0 +1,308 @@
+            'Shop' => [
+                'ShopLocale' => $this->localeFacade->getCurrentLocaleName(),
+                'ShopUrl' => $this->config->getHostYves(),
+            ],
+        ];
+
+        foreach ($orderTransfer->getItems() as $item) {
+            $payload['OrderItem'][] = [
+                'Name' => $item->getName(),
+                'Sku' => $item->getSku(),
+                'Image' => $this->getItemImageLink($item->getImages()),
+                'DeepLink' => $this->getDeepLink($item, $locale),
+                'Price' => $this->getFormattedPriceFromInt($item->getRefundableAmount()),
+                'Quantity' => $item->getQuantity(),
+                'Sum' => $this->getFormattedPriceFromInt($item->getSumPriceToPayAggregation()),
+                'OriginalPrice' => $this->getFormattedPriceFromInt($item->getUnitPrice()),
+                'TaxAmount' => $this->getFormattedPriceFromInt($item->getUnitTaxAmount()),
+                'TaxRate' => $item->getTaxRate(),
+                'Discount' => $this->getFormattedPriceFromInt($item->getUnitDiscountAmountAggregation()),
+            ];
+        }
+
+        return $payload;
+    }
+
+    /**
+     * @return string
+     */
+    abstract protected function getEvent(): string;
+
+    /**
+     * @param \ArrayObject $images
+     *
+     * @return string
+     */
+    protected function getItemImageLink(ArrayObject $images): string
+    {
+        /**
+         * @var \Generated\Shared\Transfer\ProductImageTransfer $image
+         */
+        foreach ($images as $image) {
+            return $image->getExternalUrlSmall();
+        }
+
+        return '';
+    }
+
+    /**
+     * @param \ArrayObject $methods
+     * @param \ArrayObject $expenses
+     *
+     * @return array
+     */
+    protected function getOrderDeliveryInfo(ArrayObject $methods, ArrayObject $expenses): array
+    {
+        $result = [];
+
+        /**
+         * @var \Generated\Shared\Transfer\ShipmentMethodTransfer $method
+         */
+        foreach ($methods as $method) {
+            $result[] = [
+                'DeliveryMethod' => $method->getName(),
+                'DeliveryService' => $method->getCarrierName(),
+                'DeliveryCosts' => $this->getDeliveryCosts($expenses),
+                'ShippingDate' => $this->dateTimeService->formatDateTime((new DateTime())::createFromFormat('U', $method->getDeliveryTime())),
+                'MultiDelivery' => "No",
+            ];
+        }
+
+        return $result;
+    }
+
+    /**
+     * @param \ArrayObject $methods
+     *
+     * @return array
+     */
+    protected function getPaymentMethodInfo(ArrayObject $methods): array
+    {
+        $result = [];
+
+        /**
+         * @var \Generated\Shared\Transfer\PaymentTransfer $method
+         */
+        foreach ($methods as $method) {
+            $result[] = [
+                'PaymentMethod' => $method->getPaymentMethod(),
```

Comment:
magic strings.

Reply:
It's okay for mappers. It really depends on projects needed fields.